### ISSUE TITLE
company_name search does not always return results

### ISSUE THREAD

**Issue by jerico** [younginnovations/resourcecontracts-api#7]
I'm listing companies from company_name key. Other company name work, some don't.

If I try "company_name=OceanaGold (Philippines), Inc. (Contractor/Operator)", it does not return any result. It should return 1 contract associated with that company.

http://api.resourcecontracts.org/contracts/search?from=0&per_page=1000&group=metadata&country=ph&company_name=OceanaGold%20(Philippines),%20Inc.%20(Contractor%2FOperator)&

**Comment by anderspeders** [comment]
@anjesh Can you follow up here?

**Comment by anjesh** [comment]
I think this has to do with the punctuation characters. We need to explore a bit on the elasticsearch indexing features. I remove comma from one of the companies http://contracts.ph-eiti.org/contract/101 and now it's appearing in the results. http://contracts.ph-eiti.org/search?q=&year=&resource=&company_name=Adnama%20Mining%20Resources%20Incorporated

**Comment by anderspeders** [comment]
also - would it make sense to push this repo to NRGI/RC as well?

**Comment by anjesh** [comment]
Yes definitely. I do have plans to move this and 2 more repo (subsite and elasticsearch) to NRGI github.

**Comment by anderspeders** [comment]
Great, thanks.

**Comment by anjesh** [comment]
Presence of comma in the company name is barring the contracts from appearing in the results. One thing we could do here is remove the comma from the company name, as you may see the result appears when the comma is removed. 

![image](https://cloud.githubusercontent.com/assets/120488/10608522/12dd6132-775f-11e5-8a0a-3d4aba8f1552.png)

![image](https://cloud.githubusercontent.com/assets/120488/10608538/30c8846a-775f-11e5-9b82-12634afedbe3.png)

However the system doesn't differentiate the supporting contracts from the main contracts as of now (except that there's relationship), so when the company names are returned from the API, it gives all the company names including the ones from the supporting ones as well. And I see that Jerico has hidden all the supporting contracts, perhaps using text pattern "annex?" search. If the company name in supporting documents is different from principal document, then we will see the company name whereas it won't display any results. 

@charlesyoung

**Comment by charlesyoung** [comment]
Yep is a comma issue, same with Forum Exploration, Incorporated.

There shouldn't a comma in the name so can we build some form of validation when capturing the company name in the admin module? 

I will for now update the contracts and annex's linked with a company that has a comma in the name like I just did for Far Southeast Gold Resources Incorporated.

**Comment by charlesyoung** [comment]
Busy updating, found more issues. 

Another issue is that the system crashes when the name includes a bracket (OceanaGold (Philippines), Incorporated - FTAA No. 001, 1994). I have asked Jerico to speak to Joy to update. 

Also doesn't support hyphens (Rapu-Rapu Minerals, Incorporated - MPSA No. 163-2000-V, 2000) which is a problem because in this example the company name needs a hyphen.

**Comment by charlesyoung** [comment]
Scrape above, was related to the site going down.

**Comment by charlesyoung** [comment]
Interesting that the contract name isn't updated when I remove the comma. 

![image](https://cloud.githubusercontent.com/assets/13888372/10636836/8e291824-7802-11e5-88bc-8b204a657980.png)
